### ISSUE TITLE
Entity Stacking By Chunk

### ISSUE THREAD

**Issue by ElysianPvP** [Imanity-Software/ImanitySpigot3-git#220]
### Is your feature request related to a problem?

I'm developing a Factions server that was going to develop entity stacking in part of one of my plugins. The way we use entity stacking Factions is by the chunk. So when a spawner spawns entities inside of a chunk, they typically all go into the same stack, where ever it is within the chunk, without spawning an extra entity. 

Right now, the only option is "stackRadius" and that doesn't do the job I'm searching for. In Factions you divide the base up to to your players and give them what is called their "Boxes". In these "Boxes", they are split by chunks and they basically only have access to their own chunks. With the method of "stackRadius", it currently can steal entities from one chunk to the next. That shouldn't happen at all for what Factions servers need.

### Describe the solution you'd like.

The solution I'd like is for a boolean option "stackByChunk" where immediately when an entity spawns or should spawn, it locates if there is already a stack of that entity type inside the chunk and stacks directly into them. This should override the "stackRadius" option and make that obsolete.  Instead causing entities to stack directly into their stack by chunk and not by radius.

So basically this is how a farm looks like.
![image](https://github.com/Imanity-Software/ImanitySpigot3-git/assets/118420609/a883c61b-90ca-4f40-8bab-6c6ec1bf15ab)

However, it's much better for Factions for instead of multiple of those entities, to instead have them instantly go straight to the stack.

### Describe alternatives you've considered.

I developed my own plugin in the past and there are a few public plugins available. Haven't found anything super optimized. ImanitySpigot3 already has the option for mob stacking and selecting by chunk I don't think would be much of an issue.

### Agreements

- [X] I have searched for and ensured there isn't already an open issue regarding this.
- [X] I have ensured the feature I'm requesting isn't already in the latest ImanitySpigot3.
